src/lib/components/SideNav/index.js

```diff
@@ -0,0 +1,195 @@
+            href="/"
+            class="gc-analytics-event"
+            data-category="Site-Wide Custom Events"
+            data-label="Site logo"
+          >
+            <img
+              class="web-side-nav__logo"
+              src="/images/lockup.svg"
+              alt="web.dev"
+            />
+          </a>
+        </div>
+        ${this.prerenderedChildren_}
+      </nav>
+    `;
+  }
+
+  firstUpdated() {
+    this.sideNavContainerEl = this.querySelector(".web-side-nav__container");
+    this.addEventListeners();
+    store.subscribe(this.onStateChanged);
+    this.onStateChanged();
+  }
+
+  addEventListeners() {
+    this.addEventListener("click", this.onCloseSideNav);
+    this.addEventListener("touchstart", this.onTouchStart, {passive: true});
+    this.addEventListener("touchmove", this.onTouchMove, {passive: true});
+    this.addEventListener("touchend", this.onTouchEnd);
+  }
+
+  onStateChanged() {
+    const {isSideNavExpanded} = store.getState();
+    if (isSideNavExpanded === this.expanded) {
+      return;
+    }
+
+    this.expanded = isSideNavExpanded;
+  }
+
+  onTouchStart(e) {
+    if (!this.expanded) {
+      return;
+    }
+
+    this.startX_ = e.touches[0].pageX;
+    this.currentX_ = this.startX_;
+
+    this.touchingSideNav_ = true;
+    requestAnimationFrame(this.drag);
+  }
+
+  onTouchMove(e) {
+    if (!this.touchingSideNav_) {
+      return;
+    }
+
+    this.currentX_ = e.touches[0].pageX;
+  }
+
+  onTouchEnd(e) {
+    if (!this.touchingSideNav_) {
+      return;
+    }
+
+    this.touchingSideNav_ = false;
+
+    const translateX = Math.min(0, this.currentX_ - this.startX_);
+    this.sideNavContainerEl.style.transform = "";
+
+    if (translateX < 0) {
+      this.onCloseSideNav();
+    }
+  }
+
+  drag() {
+    if (!this.touchingSideNav_) {
+      return;
+    }
+
+    requestAnimationFrame(this.drag);
+
+    const translateX = Math.min(0, this.currentX_ - this.startX_);
+    this.sideNavContainerEl.style.transform = `translateX(${translateX}px)`;
+  }
+
+  onBlockClicks(e) {
+    e.stopPropagation();
+  }
+
+  onTransitionEnd() {
+    this.animatable = false;
+  }
+
+  onCloseSideNav() {
+    // It's important to call the closeSideNav() action here instead of just
+    // setting expanded = false.
+    // The closeSideNav() action will inform other page elements that they
+    // should un-inert themselves.
+    closeSideNav();
+  }
+
+  set expanded(val) {
+    if (this.expanded_ === val) {
+      return;
+    }
+
+    const oldVal = this.expanded_;
+    this.animatable = true;
+    this.addEventListener("transitionend", this.onTransitionEnd, {once: true});
+    this.expanded_ = val;
+    if (this.expanded_) {
+      this.inert = false;
+    } else {
+      this.inert = true;
+    }
+    this.requestUpdate("expanded", oldVal);
+  }
+
+  get expanded() {
+    return this.expanded_;
+  }
+
+  disconnectedCallback() {
+    super.disconnectedCallback();
+    store.unsubscribe(this.onStateChanged);
```

Comment:
The way I read the docs for `firstUpdated`, it seems to be one-time, not after every addition to the page.

https://lit-element.polymer-project.org/guide/lifecycle#firstupdated

Maybe move the subscribe call to `connectedCallback`?